pandas/core/frame.py

```diff
@@ -1141,7 +1142,17 @@ def __len__(self) -> int:
         """
         return len(self.index)
 
-    def dot(self, other):
+    # pandas/core/frame.py:1146: error: Overloaded function signatures 1 and 2
+    # overlap with incompatible return types  [misc]
+    @overload
+    def dot(self, other: Series) -> Series:  # type: ignore[misc]
+        ...
+
+    @overload
+    def dot(self, other: Union[DataFrame, Index, ArrayLike]) -> DataFrame:
+        ...
+
+    def dot(self, other: Union[AnyArrayLike, FrameOrSeriesUnion]) -> FrameOrSeriesUnion:
```

Comment:
@arw2019 just FYI

`error: Overloaded function implementation cannot satisfy signature 2 due to inconsistencies in how they use type variables  [misc]`

when we add numpy types.

the problem is the Series in AnyArrayLike could be a Series subclass.

I would like to change AnyArrayLike to a Union instead of a TypeVar to avoid many issues like this

good news though. the ignore above is not needed.